deps/wazuh_testing/wazuh_testing/tools/system.py

```diff
@@ -338,6 +444,153 @@ def configure_local_internal_options(self, local_internal_options: dict):
                 replace = replace + internal_option
             self.modify_file_content(target_host, WAZUH_LOCAL_INTERNAL_OPTIONS, replace)
 
+    def download_file(self, host, url, dest_path, mode='755'):
+        """
+        Downloads a file from the specified URL to the destination path on the specified host.
+
+        Args:
+            host (str): The target host on which to download the file.
+            url (str): The URL of the file to be downloaded.
+            dest_path (str): The destination path where the file will be saved on the host.
+            mode (str, optional): The file permissions mode. Defaults to '755'.
+
+        Returns:
+            dict: Ansible result of the download operation.
+
+        Example:
+            host_manager.download_file('my_host', 'http://example.com/path/file.conf', '/etc/foo.conf', mode='0440')
+        """
+        result = self.get_host(host).ansible("get_url", f"url={url} dest={dest_path} mode={mode}", check=False)
+
+        return result
+
+    def install_package(self, host, url, system='ubuntu'):
+        """
+        Installs a package on the specified host.
+
+        Args:
+            host (str): The target host on which to install the package.
+            url (str): The URL or name of the package to be installed.
+            system (str, optional): The operating system type. Defaults to 'ubuntu'.
+                Supported values: 'windows', 'ubuntu', 'centos'.
+
+        Returns:
+            Dict: Testinfra Ansible Response of the operation 
+
+        Example:
+            host_manager.install_package('my_host', 'http://example.com/package.deb', system='ubuntu')
+        """
+        result = False
+
+        if system =='windows':
+            result = self.get_host(host).ansible("win_package", f"path={url} arguments=/S", check=False)
+        elif system == 'ubuntu':
+            result = self.get_host(host).ansible("apt", f"deb={url}", check=False)
+            if result['changed'] == True and result['stderr'] == '':
+                result = True
+        elif system == 'centos':
+            result = self.get_host(host).ansible("yum", f"name={url} state=present sslverify=false disable_gpg_check=True", check=False)
+            if 'rc' in result and result['rc'] == 0 and result['changed'] == True:
+                result = True
+
+        return result
+
+    def get_master_ip(self):
+        """
+        Retrieves the IP address of the master node from the inventory.
+
+        Returns:
+            str: The IP address of the master node, or None if not found.
+
+        Example:
+            master_ip = host_manager.get_master_ip()
+        """
+        master_ip = None
+
+        for manager in self.get_group_hosts('manager'):
+            if 'type' in self.get_host_variables(manager) and \
+                self.get_host_variables(manager)['type'] == 'master':
+                master_ip = self.get_host_variables(manager)['ip']
```

Comment:
Is it not overwritten if it later checks a worker?

Reply:
No, due it is only one host that has the type master. However, I include a break to improve performance

Done 372138c20